addons/sale_product_matrix/models/sale_order.py

```diff
@@ -115,18 +115,9 @@ def _apply_grid(self):
                         product_uom_qty=qty,
                         product_no_variant_attribute_value_ids=no_variant_attribute_values.ids)
                     ))
-            if product_ids:
-                res = False
-                if new_lines:
-                    # Add new SO lines
-                    self.update(dict(order_line=new_lines))
-
-                # Recompute prices for new/modified lines
-                for line in self.order_line.filtered(lambda line: line.product_id.id in product_ids):
-                    res = line.product_id_change() or res
-                    line._onchange_discount()
-                    line._onchange_product_id_set_customer_lead()
-                return res
+            if product_ids and new_lines:
```

Comment:
```suggestion
            if new_lines:
```

And the product_ids variable (& logic) can be removed.